src/controllers/viewTripHistory.js

```diff
@@ -0,0 +1,28 @@
+
+import getDataFromToken from '../helper/tokenToData';
+import findTrip, { displayNumberOfTrips } from '../services/tripHistorySearch';
+
+
+export const getTripHistory = async (req, res, next) => {
+  const decoded = await getDataFromToken(req, res, next);
+  
+    const userid = decoded.id.toString();
+    const offset = req.query.page;
+    const limit = req.query.limit;
+    const location = req.params.location;
+    let query = { userId: userid };
+    findTrip(res, query, location, offset, limit);
```

Comment:
@edgar929 you don't need to pass the res object to findTrip function. FindTrip function should only get parameters/arguments that it only needs to find the trip info. FindTrip function should return the data i.e. trip info and then getTripHistory returns the response to the user. It's not a good idea to keep passing down req & res object. Kindly update elsewhere you've passed the res object down to another function.